### PR TITLE
Add FXIOS-9987 Add Tab unit tests for isActive and isInactive

### PR DESCRIPTION
**Pull request by ih-codes**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-TODO)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21930)

## :bulb: Description
This PR adds (and updates) unit tests related to a Tab's `isActive` and `isInactive` state. Follow up to PR #21913.

### Discussion

Interestingly, after I looked into this a bit more, I realized `lastExecutedTime` and `firstCreatedTime` are never set to `nil`. In fact, the init was giving them a current Date default time. 

So, I refactored both properties to be non-optional, which simplified code in a number of places. This also means the `isActive` and `isInactive` logic is a little less complicated (we don't have to test for the edge case where no execution time exists).

Hopefully there will be less room for error in the future as well.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/TabManagement/Tab.swift [hunk 1]
```diff
@@ -415,16 +419,12 @@ class Tab: NSObject, ThemeApplicable {
         if UserDefaults.standard.bool(forKey: PrefsKeys.FasterInactiveTabsOverride) {
             inactiveDate = Calendar.current.date(byAdding: .second, value: -10, to: currentDate) ?? Date()
         } else {
+            // FIXME Is there a reason we use noon of the current day instead of the exact time, when calculating -14 days?
             inactiveDate = Calendar.current.date(byAdding: .day, value: -14, to: currentDate.noon) ?? Date()
```
Comment (ih-codes) [inline firefox-ios/Client/TabManagement/Tab.swift:423]: Just wanted to call this out. Is there a reason for this extra logic to check based on `noon` of the current day, vs. the actual time? 🤔  After 14 days, what's a couple hours difference...?

Comment (dataports) [reply firefox-ios/Client/TabManagement/Tab.swift:423]: Maybe just for consistency for debugging (in cases across timezones or some other edge case?) someone thought it would be better? 
I also imagine it would be easier to look at a bunch of lines of data of multiple inactive tabs if you only had to account for the date and not the time? Really just guessing though, Orla might know the actual answer to this 😆 
Ex: 
12/2/2024 12:00:00, 
13/2/2024 12:00:00

vs 
12/2/2024 01:22:33, 
13/2/2024 23:02:13

Comment (OrlaM) [reply firefox-ios/Client/TabManagement/Tab.swift:423]: I have no idea what the original thinking was here 🙈

**Review by OrlaM — APPROVED** [review]
Great work!

File: firefox-ios/Client/TabManagement/Tab.swift (hunk 1 above)
Comment (ih-codes) [reply firefox-ios/Client/TabManagement/Tab.swift:423]: Huh, interesting! 😂  I guess my vote is on removing it since we don't seem to have a good reason to be this specific on the time, but I can leave it if we're not confident of the potential impact.

Comment (dataports) [reply firefox-ios/Client/TabManagement/Tab.swift:423]: I'd say if this info isn't feeding into any telemetry probably safe to change it, if it is feeding into telemetry then keep it. 👍 I don't remember it feeding into anything when I looked yesterday.

**Review by dataports — COMMENTED** [review]
This and the previous PR are so great, thank you so much for helping with the inactive tabs stuff and cleaning up these files so much ❤️

File: firefox-ios/Client/TabManagement/Tab.swift (hunk 1 above)
Comment (ih-codes) [reply firefox-ios/Client/TabManagement/Tab.swift:423]: Thanks Sophie, I didn't think of that possibility. 🤔  I'll make a [spike task](https://mozilla-hub.atlassian.net/browse/FXIOS-10005) to look into this more later since it's just a nit thing.

**Comment by ih-codes** [comment]
@mergify backport release/v131
